plugins/methodinfo/src/main/java/org/qbicc/plugin/methodinfo/MethodDataEmitter.java

```diff
@@ -97,9 +98,15 @@ private int createSourceCodeInfo(CompilationContext ctxt, MethodData methodData,
         int sourceCodeIndex;
         ExecutableElement element = node.getElement();
         if (node.getCallSite() == null) {
+            if (element.hasAllModifiersOf(ClassFile.I_ACC_HIDDEN)) {
```

Comment:
Another way to skip methods annotated as `Hidden` would be to store the `modifiers` value in method data and do the check for `ClassFile.I_ACC_HIDDEN` in `JavaStackWalker#skipFrame`. This would result in extra 4 bytes for every entry in `qbicc_method_info_table`.

Reply:
I checked up `qbicc_method_info` structure and it appears there is already a 4 byte of padding at the end to align the struct to 64-bits. I think storing method `modifiers` in method data would be a cleaner solution. The overhead would be the method data for hidden methods which would not be needed for stack trace. Not sure if it would be useful for any other purpose. In any case I expect that overhead in memory should be very low and hence acceptable.

Reply:
I like that solution; it could be useful for other purposes as well.

Reply:
Updated the commit to follow this.